x509util/certificate.go

```diff
@@ -79,15 +80,25 @@ func (c *Certificate) GetCertificate() *x509.Certificate {
 	cert.PublicKey = c.PublicKey
 	cert.PublicKeyAlgorithm = c.PublicKeyAlgorithm
 
-	// SANs are directly converted.
-	cert.DNSNames = c.DNSNames
-	cert.EmailAddresses = c.EmailAddresses
-	cert.IPAddresses = c.IPAddresses
-	cert.URIs = c.URIs
+	if c.hasExtendedSANs() && !c.hasExtension(subjectAlternativeNameOID) {
```

Comment:
Nice, only use createSubjectAltNameExtension if it has SANs non supported by Go.